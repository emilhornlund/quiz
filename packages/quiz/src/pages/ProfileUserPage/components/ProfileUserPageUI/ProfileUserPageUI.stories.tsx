import type { Meta, StoryObj } from '@storybook/react'
import { withRouter } from 'storybook-addon-remix-react-router'

import { withMockAuth } from '../../../../../.storybook/mockAuthContext.tsx'

import ProfileUserPageUI from './ProfileUserPageUI.tsx'

const meta = {
  title: 'Pages/ProfileUserPage',
  component: ProfileUserPageUI,
  decorators: [withRouter, withMockAuth],
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof ProfileUserPageUI>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {
    values: {
      email: '',
      unverifiedEmail: undefined,
      givenName: '',
      familyName: '',
      defaultNickname: '',
    },
    loading: false,
    loadingPassword: false,
    onChange: () => undefined,
    onChangePassword: () => undefined,
    onClickResendVerificationEmail: () => undefined,
  },
} satisfies Story
