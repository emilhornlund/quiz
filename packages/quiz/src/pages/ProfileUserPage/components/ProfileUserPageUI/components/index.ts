export type {
  UpdateUserDetailsFormFields,
  UserDetailsFormProps,
} from './UserDetailsForm'
export type {
  UpdateUserPasswordFormFields,
  UserPasswordFormProps,
} from './UserPasswordForm'
export { default as UserDetailsForm } from './UserDetailsForm'
export { default as UserPasswordForm } from './UserPasswordForm'
