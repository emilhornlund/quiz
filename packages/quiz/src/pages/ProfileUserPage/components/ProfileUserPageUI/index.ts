export type {
  UpdateUserProfileFormFields,
  ProfileUserPageUIProps,
} from './ProfileUserPageUI.tsx'
export { default } from './ProfileUserPageUI.tsx'
