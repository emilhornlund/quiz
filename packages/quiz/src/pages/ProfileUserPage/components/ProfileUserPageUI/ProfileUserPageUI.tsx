import { AuthProvider } from '@quiz/common'
import React, { FC } from 'react'

import { LegacyInfoCard, Page, PageDivider } from '../../../../components'

import {
  UpdateUserDetailsFormFields,
  UpdateUserPasswordFormFields,
  UserDetailsForm,
  UserPasswordForm,
} from './components'

export interface ProfileUserPageUIProps {
  authProvider: AuthProvider
  values: UpdateUserDetailsFormFields
  loading: boolean
  loadingPassword: boolean
  onChange: (request: UpdateUserDetailsFormFields) => void
  onChangePassword: (request: UpdateUserPasswordFormFields) => void
  onClickResendVerificationEmail: () => void
}

const ProfileUserPageUI: FC<ProfileUserPageUIProps> = ({
  authProvider,
  values,
  loading,
  loadingPassword,
  onChange,
  onChangePassword,
  onClickResendVerificationEmail,
}) => (
  <Page align="start" discover profile>
    <LegacyInfoCard />
    <UserDetailsForm
      authProvider={authProvider}
      values={values}
      loading={loading}
      onChange={onChange}
      onClickResendVerificationEmail={onClickResendVerificationEmail}
    />
    {authProvider === AuthProvider.Local && (
      <>
        <PageDivider />
        <UserPasswordForm
          loading={loadingPassword}
          onChange={onChangePassword}
        />
      </>
    )}
  </Page>
)

export default ProfileUserPageUI
