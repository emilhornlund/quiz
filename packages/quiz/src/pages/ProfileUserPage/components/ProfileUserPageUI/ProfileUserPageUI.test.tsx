import { render } from '@testing-library/react'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { describe, expect, it } from 'vitest'

import ProfileUserPageUI from './ProfileUserPageUI.tsx'

describe('ProfileUserPageUI', () => {
  it('should render ProfileUserPageUI', async () => {
    const { container } = render(
      <MemoryRouter>
        <ProfileUserPageUI
          values={{
            email: '',
            givenName: '',
            familyName: '',
            defaultNickname: '',
          }}
          loading={false}
          loadingPassword={false}
          onChange={() => undefined}
          onChangePassword={() => undefined}
        />
      </MemoryRouter>,
    )

    expect(container).toMatchSnapshot()
  })
})
