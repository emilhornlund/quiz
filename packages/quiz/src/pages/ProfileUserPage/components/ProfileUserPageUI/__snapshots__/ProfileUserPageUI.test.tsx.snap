// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileUserPageUI > should render ProfileUserPageUI 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <a
          data-discover="true"
          href="/auth/login"
        >
          Login
        </a>
      </div>
    </div>
    <div
      class="content fullWidth startAlign"
    >
      <div
        class="typography subtitle full"
      >
        Shape your quiz identity
      </div>
      <div
        class="typography text medium"
      >
        Update your personal details to keep your profile up to date. Your information helps personalize your quiz experience and lets others recognize you during games.
      </div>
      <form
        class="form"
      >
        <div
          class="inputContainer"
        >
          <div
            class="textFieldInputContainer textFieldInputKindPrimary"
          >
            <input
              class="textfield"
              data-testid="test-email-textfield"
              id="email"
              maxlength="128"
              minlength="6"
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="inputContainer"
        >
          <div
            class="textFieldInputContainer textFieldInputKindPrimary"
          >
            <input
              class="textfield"
              data-testid="test-givenName-textfield"
              id="givenName"
              maxlength="64"
              minlength="1"
              name="givenName"
              placeholder="Given Name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="inputContainer"
        >
          <div
            class="textFieldInputContainer textFieldInputKindPrimary"
          >
            <input
              class="textfield"
              data-testid="test-familyName-textfield"
              id="familyName"
              maxlength="64"
              minlength="1"
              name="familyName"
              placeholder="Family Name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="inputContainer"
        >
          <div
            class="textFieldInputContainer textFieldInputKindPrimary"
          >
            <input
              class="textfield"
              data-testid="test-defaultNickname-textfield"
              id="defaultNickname"
              maxlength="20"
              minlength="2"
              name="defaultNickname"
              placeholder="Default Nickname"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="buttonInputContainer buttonInputKindCallToAction"
        >
          <button
            data-testid="test-update-user-button-button"
            disabled=""
            id="update-user-button"
            name="update-user-button"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-floppy-disk "
              data-icon="floppy-disk"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M64 32C28.7 32 0 60.7 0 96L0 416c0 35.3 28.7 64 64 64l320 0c35.3 0 64-28.7 64-64l0-242.7c0-17-6.7-33.3-18.7-45.3L352 50.7C340 38.7 323.7 32 306.7 32L64 32zm0 96c0-17.7 14.3-32 32-32l192 0c17.7 0 32 14.3 32 32l0 64c0 17.7-14.3 32-32 32L96 224c-17.7 0-32-14.3-32-32l0-64zM224 288a64 64 0 1 1 0 128 64 64 0 1 1 0-128z"
                fill="currentColor"
              />
            </svg>
            <span>
              Save!
            </span>
          </button>
        </div>
      </form>
      <div
        class="pageDivider"
      />
      <div
        class="typography subtitle full"
      >
        Upgrade Your Password
      </div>
      <div
        class="typography text medium"
      >
        Enhance your account security with a fresh passwordâ€”verify your current one, choose a new one, and confirm to complete the update.
      </div>
      <form
        class="form"
      >
        <div
          class="inputContainer"
        >
          <div
            class="textFieldInputContainer textFieldInputKindPrimary"
          >
            <input
              class="textfield"
              data-testid="test-oldPassword-textfield"
              id="oldPassword"
              maxlength="128"
              minlength="8"
              name="oldPassword"
              placeholder="Old Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="inputContainer"
        >
          <div
            class="textFieldInputContainer textFieldInputKindPrimary"
          >
            <input
              class="textfield"
              data-testid="test-newPassword-textfield"
              id="newPassword"
              maxlength="128"
              minlength="8"
              name="newPassword"
              placeholder="New Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="inputContainer"
        >
          <div
            class="textFieldInputContainer textFieldInputKindPrimary"
          >
            <input
              class="textfield"
              data-testid="test-confirmPassword-textfield"
              id="confirmPassword"
              maxlength="128"
              minlength="8"
              name="confirmPassword"
              placeholder="Confirm Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="buttonInputContainer buttonInputKindCallToAction"
        >
          <button
            data-testid="test-update-password-button-button"
            disabled=""
            id="update-password-button"
            name="update-password-button"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-lock "
              data-icon="lock"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M144 144l0 48 160 0 0-48c0-44.2-35.8-80-80-80s-80 35.8-80 80zM80 192l0-48C80 64.5 144.5 0 224 0s144 64.5 144 144l0 48 16 0c35.3 0 64 28.7 64 64l0 192c0 35.3-28.7 64-64 64L64 512c-35.3 0-64-28.7-64-64L0 256c0-35.3 28.7-64 64-64l16 0z"
                fill="currentColor"
              />
            </svg>
            <span>
              Lock It Down!
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
